### PR TITLE
feat: pending transactions management

### LINKED ISSUE #396: Cancel transaction support
https://github.com/ethersphere/bee/pull/2212/

### PR DESCRIPTION
**Pull request by AuHau**
Closes #396

### CODE REVIEW

File: src/bee-debug.ts [hunk 1]
```diff
@@ -380,4 +390,49 @@ export class BeeDebug {
...
+  async getPendingTransaction(transactionHash: TransactionHash | string): Promise<TransactionInfo> {
+    assertTransactionHash(transactionHash)
+
+    return transactions.getTransaction(this.url, transactionHash)
+  }
+
+  /**
+   * Rebroadcast already created transaction
+   * @param transactionHash
+   */
```
Comment (Cafe137) [inline src/bee-debug.ts:414]: I think the JSDoc should explain what rebroadcast is good for, when to use it.

The function is called `rebroadcastPending` but the JSDoc says _Rebroadcast already created_, not sure if this is intentional, but it is a bit confusing to me. 🙂

Comment (AuHau) [reply src/bee-debug.ts:414]: Yeah agree. I'm just not sure exactly what is it about so choose not to expand on that 😅

Maybe @vojtechsimetka could you please help me out with what should be written here?

Comment (nugaon) [reply src/bee-debug.ts:414]: For me the PR's code is good, but this part also little bit vague for me what this function does.

Comment (vojtechsimetka) [reply src/bee-debug.ts:414]: From the Ethereum perspective this _Rebroadcasts already created transaction_ also called _pending transaction_ (as it is a transaction that has been before send to the network but is not yet included in any block). Not sure what is the confusion here :) . In the other function you have comment _Rebroadcast existing transaction_ which imho is also OK.

**Review by vojtechsimetka — APPROVED** [review]

**Review by nugaon — APPROVED** [review]

**Pull request merged**